src/main/java/net/dries007/tfc/common/recipes/SimpleCraftMatrix.java

```diff
@@ -0,0 +1,270 @@
+/*
+ * Licensed under the EUPL, Version 1.2.
+ * You may obtain a copy of the Licence at:
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ */
+
+package net.dries007.tfc.common.recipes;
+
+import java.util.Arrays;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonSyntaxException;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.JSONUtils;
+
+import net.dries007.tfc.TerraFirmaCraft;
+
+/**
+ * A simple craft matrix for knapping / leather or clay working
+ *
+ * @author AlcatrazEscapee
+ */
+public class SimpleCraftMatrix
+{
+    private static final int MAX_WIDTH = 5;
+    private static final int MAX_HEIGHT = 5;
+    private static final int MAX_AREA = MAX_WIDTH * MAX_HEIGHT;
+
+    private static void logMatrix(boolean[] matrix)
+    {
+        StringBuilder b = new StringBuilder();
+        for (boolean m : matrix) b.append(m ? "X" : " ");
+        TerraFirmaCraft.LOGGER.debug("Matrix: {" + b + "}");
```

Comment:
Don't use the `TerraFirmaCraft` logger, have an instance per class if needed (or, because I doubt this method still needs to exist, yeet it)